src/replication.c

```diff
@@ -2253,6 +2259,8 @@ handle_mgmt_conn_error(replica_t *r, int sfd, struct epoll_event *events, int ev
 				if (r_ev == r) {
 					TAILQ_REMOVE(&r->spec->rwaitq, r, r_waitnext);
 					r->conn_closed++;
+					if (r->io_resp_hdr)
```

Comment:
set io_resp_hdr to NULL here, and also, in handle_data_conn_error() (with lock)

Reply:
ok